### PR TITLE
SDIT-1113 Respond to Syscon security report

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/test/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/HealthResourceTest.kt [hunk 1]
```diff
@@ -2,21 +2,11 @@ package uk.gov.justice.digital.hmpps.cmd.api.controllers
...
-import org.springframework.test.context.ActiveProfiles
-import org.springframework.test.context.junit.jupiter.SpringExtension
 
-@ExtendWith(SpringExtension::class)
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
-@ActiveProfiles(value = ["test"])
-class HealthResourceTest(@Autowired val testRestTemplate: TestRestTemplate) {
-
-  val jsonTester = BasicJsonTester(this.javaClass)
+class HealthResourceTest(@Autowired val testRestTemplate: TestRestTemplate) : ResourceTest() {
```
Comment (steverendell) [inline src/test/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/HealthResourceTest.kt:9]: pulled out a common base class

File: src/test/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/NotificationControllerGetIntegrationTest.kt [hunk 2]
```diff
@@ -182,14 +172,14 @@ class NotificationControllerGetIntegrationTest(
...
+      entityBuilder.entityWithJwtAuthorisation(user, CMD_ROLE),
       String::class.java,
     )
 
   companion object {
 
     private const val A_USER = "API_TEST_USER"
     private const val A_USER_NO_DATA = "API_TEST_USER_NO_DATA"
-    private val NO_ROLES = listOf<String>()
+    private val CMD_ROLE = listOf("ROLE_PRISON")
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/NotificationControllerGetIntegrationTest.kt:183]: needs renaming too?

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/test/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/NotificationControllerGetIntegrationTest.kt (hunk 2 above)
Comment (steverendell) [reply src/test/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/NotificationControllerGetIntegrationTest.kt:183]: ok will do next time 🙂

File: src/test/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/ResourceSecurityTest.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    "GET /swagger-ui.html",
+    "GET /v3/api-docs",
+    "GET /v3/api-docs/swagger-config",
+    " /error",
+  )
+
+  @Test
+  fun `Ensure all endpoints protected with PreAuthorize`() {
+    // need to exclude any that are forbidden in helm configuration
+    val exclusions = File("helm_deploy").walk().filter { it.name.equals("values.yaml") }.flatMap { file ->
```
[the remark this answers is not in the document]
Comment (petergphillips) [reply src/test/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/ResourceSecurityTest.kt:27]: agreed, happy with that answer - I thought that might be the case but thought I'd better check

**Pull request merged**